src/components/dashboard/todos-by-goal/todos-by-goal-container.tsx

```diff
@@ -1,29 +1,32 @@
+"use client";
+
+import { useSuspenseInfiniteQuery } from "@tanstack/react-query";
+
+import FlagIcon from "@/assets/flag.svg";
+import { useInfiniteScroll } from "@/hooks/use-infinite-scroll";
+import { getInfiniteGoalsOptions } from "@/services/dashboard/query";
+
+import SectionContainer from "../section-container";
+import SectionTitle from "../section-title";
+
+import GoalList from "./goal-list";
+
 export default function TodosByGoalContainer() {
+  const { data, fetchNextPage, hasNextPage } = useSuspenseInfiniteQuery(
+    getInfiniteGoalsOptions({}),
+  );
+  const { ref } = useInfiniteScroll({ hasNextPage, fetchNextPage });
+
   return (
-    <section className="w-full rounded-2xl bg-white md:col-span-2 md:mt-16">
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-      <div className="h-100">hello</div>
-    </section>
+    <SectionContainer className="md:col-span-2">
+      <SectionTitle>
+        <div className="flex h-40 w-40 items-center justify-center rounded-15 bg-blue-500">
+          <FlagIcon />
+        </div>
+        <h2>최근 등록한 할 일</h2>
```

Comment:
이 부분 텍스트 변경 누락된 것 같네요!

```suggestion
        <h2 className="text-lg font-semibold text-slate-800">목표 별 할 일</h2>
```

Reply:
원래 PR 올리기 전에 보고 또 봐도 안보일 때가 있죠 ㅎㅎ 수정 감사합니다~!